### PR TITLE
document template readiness

### PR DESCRIPTION
**Pull request by jim-minter**
This will be a 3.7 feature.

@bparees @coreydaley ptal

### CODE REVIEW

**Comment by jim-minter** [comment]
Please merge after https://github.com/openshift/origin/pull/15402

File: dev_guide/templates.adoc [hunk 1]
```diff
@@ -733,6 +733,53 @@ follows:
...
+
+| `Deployment`
+| Object reports new ReplicaSet and deployment available (this honors readiness
+probe(s) defined on the object)
+| Object reports Progressing condition as Failed
+
+|`DeploymentConfig`
+| Object reports new ReplicationController and deployment available (this
+honors readiness probe(s) defined on the object)
+| Object reports Progressing condition as Failed
```
Comment (bparees) [inline dev_guide/templates.adoc]: s/Failed/False/?

(same for the others?)

File: dev_guide/templates.adoc [hunk 2]
```diff
@@ -733,6 +733,62 @@ follows:
...
+is welcomed.
+
+To use this feature, a template author should mark one or more objects of kind
+`Build`, `BuildConfig`, `Deployment`, `DeploymentConfig`, `Job` or `StatefulSet`
+in a template with the annotation
+`"template.alpha.openshift.io/wait-for-ready": "true"`.  Template instantiation
+will not complete until all objects marked with the above annotation report
+ready.  Similarly, if any of the annotated objects report failed, or if the
+template fails to become ready within a fixed timeout of one hour, the template
+instantiation will fail.
```
Comment (bparees) [inline dev_guide/templates.adoc]: can you add a sample?  I worry people are not going to put the annotation in the right place otherwise.

**Review by bparees — CHANGES_REQUESTED** [review]

File: dev_guide/templates.adoc (hunk 2 above)
Comment (jim-minter) [inline dev_guide/templates.adoc]: @bparees pushed.

**Comment by bparees** [comment]
lgtm, and #15402 has merged.  @openshift/team-documentation ptal/merge.

**Comment by bmcelvee** [comment]
Thanks, @jim-minter and @bparees! LGTM. I'll merge and make some minor edits in a follow-up PR.

**Comment by bmcelvee** [comment]
[rev_history]
|xref:../dev_guide/templates.adoc#dev-guide-templates[Templates]
|Added xref:../dev_guide/templates.adoc#waiting-for-template-readiness[Waiting for Template Readiness] section.
%

**Pull request merged**